html5lib/_tokenizer.py

```diff
@@ -476,21 +477,22 @@ def rcdataEndTagOpenState(self):
         return True
 
     def rcdataEndTagNameState(self):
-        appropriate = self.currentToken and self.currentToken["name"].lower() == self.temporaryBuffer.lower()
+        name = self.temporaryBuffer.translate(asciiUpper2Lower)
+        appropriate = self.currentToken and self.currentToken["name"] == name
```

Comment:
FWIW: my thinking from my previous comment was to instead change this state (and similar ones) to:

```
if data in asciiLetters:
    ...
elif data in spaceCharacters or data in ("/", ">") and self.currentToken and self.currentToken["name"].lower() == self.temporaryBuffer.lower():
    if data in spaceCharacters:
        ...
    elif data == "/":
        ...
```

etc.

At the absolute least, I think we should avoid computing `appropriate` in the `if data in asciiLetters` case? Otherwise we're doing this lower-casing after each time we add a character.

Reply:
Ok, yep - I didn't really register that after your first comment; I'll take a look at re-ordering these conditionals soon.  Thanks for detailing that a bit further.

Reply:
That's applied now; I didn't find a noticeable performance difference as a result (cpython 3.9.1), but it may be logically clearer.

Reply:
I feel like the modifications in this PR are getting blended together slightly confusingly, so I'll do a more thorough analysis soon to pick apart the individual suggestions, analyze performance for them individually, and then keep the ones that still seem useful and show performance benefits (or seem worthwhile enough to include anyway).